src/Core/WinSWCore/ServiceDescriptor.cs

```diff
@@ -95,6 +101,37 @@ public ServiceDescriptor(XmlDocument dom)
             this.dom = dom;
         }
 
+        public void ValidateAndLoadXmlSchema()
+        {
+            XmlReaderSettings settings = new XmlReaderSettings();
+
+            Assembly a = Assembly.GetExecutingAssembly();
+
+            using (Stream schemaStream = a.GetManifestResourceStream("winsw.XMLSchema.xsd"))
+            {
+                using (XmlReader schemaReader = XmlReader.Create(schemaStream))
+                {
+                    settings.Schemas.Add(null, schemaReader);
+                }
+            }
+
+            settings.ValidationType = ValidationType.Schema;
+            settings.ValidationEventHandler += new ValidationEventHandler(XmlValidationEventHandler);
+            
+            using(XmlReader reader = XmlReader.Create(BasePath + ".xml"))

```

Comment:
`BasePath` is `null` in the test because it's not assigned in `ServiceDescriptor.FromXML` but this shouldn't be a blocker. `XmlReader` can be constructed from text as well.